pyomo/contrib/doe/fim_doe.py

```diff
@@ -205,26 +208,38 @@ def __generate_flatten_timeset(self, all_info, flatten_measure_name,name_and_ind
         self.flatten_measure_timeset = flatten_measure_timeset
 
     def __model_measure_name(self):
-        '''Return pyomo string name 
+        '''Return pyomo string name
+        Need two sets of model name for sequential_finite and direct_kaug
+        When the extra index is a string, such as 'CA', k_aug asks the name to be 'C[CA,0]', while extensive form asks to be "C['CA',0]".
+        TODO: when the extra index is a number, extensive form doesn't need extra quotes. Need to find a way to identify if the extra index is a number.
         '''
         # store pyomo string name
-        measurement_names = []
+        measurement_names_ef = []
+        measurement_names_kaug = []
         # loop over measurement name
         for mname in self.flatten_measure_name:
             # check if there is extra index
             if self.ind_string in mname:
                 measure_name = mname.split(self.ind_string)[0]
                 measure_index = mname.split(self.ind_string)[1]
+
                 for tim in self.flatten_measure_timeset[mname]:
                     # get the measurement name in the model
-                    measurement_name = measure_name + '[0,' + measure_index + ',' + str(tim) + ']'
-                    measurement_names.append(measurement_name)
+                    measurement_name_kaug = measure_name + '[' + measure_index + ',' + str(tim) + ']'
+                    measurement_names_kaug.append(measurement_name_kaug)
+
+                    measurement_name_ef = measure_name + "['" + measure_index + "'," + str(tim) + "]"
+                    measurement_names_ef.append(measurement_name_ef)
+
+
             else:
                 for tim in self.flatten_measure_timeset[mname]:
                     # get the measurement name in the model
-                    measurement_name = mname + '[0,' + str(tim) + ']'
-                    measurement_names.append(measurement_name)
-        self.model_measure_name = measurement_names
+                    measurement_name = mname + '[' + str(tim) + ']'
+                    measurement_names_ef.append(measurement_name)
```

Comment:
Add a comment to explain why these are the same. I think I understand what you are doing. You want comments to make it explicit so there is no need for the reader to guess.